device/kokkos/include/traccc/kokkos/seeding/seed_finding.hpp

```diff
@@ -0,0 +1,51 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/edm/seed.hpp"
+#include "traccc/edm/spacepoint.hpp"
+#include "traccc/seeding/detail/seeding_config.hpp"
+#include "traccc/seeding/detail/spacepoint_grid.hpp"
+#include "traccc/utils/algorithm.hpp"
+
+// VecMem include(s).
+#include <vecmem/containers/data/vector_buffer.hpp>
+#include <vecmem/memory/memory_resource.hpp>
+
+// System include(s).
+#include <functional>
+
+namespace traccc::kokkos {
+
+/// Seed finding for cuda
+class seed_finding
+    : public algorithm<host_seed_collection(
+          const spacepoint_container_types::host&, const sp_grid_const_view&)> {
+
+    public:
+    /// Constructor for the cuda seed finding
+    ///
+    /// @param config is seed finder configuration parameters
+    /// @param sp_grid spacepoint grid
+    /// @param mr vecmem memory resource
+    seed_finding(const seedfinder_config& config, vecmem::memory_resource& mr);
```

Comment:
Instead of a `vecmem::memory_resource` the class should receive a [traccc::memory_resource](https://github.com/acts-project/traccc/blob/main/core/include/traccc/utils/memory_resource.hpp) object instead. That's what we've been moving all other algorithms to as well.

At the same time it should also receive a `vecmem::copy` object "from the outside". One that would be compatible with whatever memory resource(s) is/are given to it.